libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -269,8 +271,11 @@ void GCS_MAVLINK::send_battery_status(const uint8_t instance) const
     } else {
         consumed_wh = -1;
     }
-    uint8_t _percentage = -1;
-    const int8_t percentage = battery.capacity_remaining_pct(_percentage, instance) ? _percentage : -1;
+    if (battery.capacity_remaining_pct(percentage, instance)){
+        percentage = percentage>100 ? 100 : percentage;     
```

Comment:
I don't think limiting to 100 is valid, we support generators now and MPPT chargers, it is possible that the percentage would genuinely go above 100%.

We should totally fix the wrap tho. Maybe like this?
```suggestion
        percentage = MIN(percentage, INT8_MAX);     
```

Reply:
Got it. Thanks for helping. Please let me know in case of more discrepancies.

Reply:
Note: the wrap is also handled on the AP_BattMonitorBackend side as well but handling in both spots is good.